src/compojure/api/api.clj

```diff
@@ -50,14 +52,19 @@
   [& body]
   (let [[options handlers] (common/extract-parameters body false)
         options (rsc/deep-merge api-defaults options)
-        handler (apply c/routes (concat [(swagger/swagger-routes (:swagger options))] handlers))
+        raw (:raw options)
+        swagger-ui (-> (swagger/swagger-routes (:swagger options))
+                       (cond-> raw ;; if the api won't be wrapped, swagger-ui still needs to be
```

Comment:
For the first question, I'm not sure you should make assumptions about what middlewares the user has for their own routes; if there are middlewares needed for the swagger routes, those should be added for those routes specifically. For example, we don't have a json middleware that automatically detects clojure maps and converts them to json; instead we are explicit about json-encoding only the limited subset of json that we actually return from the server.

The second statement I don't understand. What do you mean, doesn't do the serialization? When I was writing this PR, I had formats set to `[:json]` or something like that, so it wasn't nil.